covalent/_workflow/lepton.py

```diff
@@ -68,18 +71,50 @@ def __init__(
         executor: Union[
             List[Union[str, "BaseExecutor"]], Union[str, "BaseExecutor"]
         ] = _DEFAULT_CONSTRAINT_VALUES["executor"],
+        deps_bash: Union[DepsBash, List, str] = _DEFAULT_CONSTRAINT_VALUES["deps"].get("bash", []),
+        deps_pip: Union[DepsPip, list] = _DEFAULT_CONSTRAINT_VALUES["deps"].get("pip", None),
+        call_before: Union[List[DepsCall], DepsCall] = _DEFAULT_CONSTRAINT_VALUES["call_before"],
+        call_after: Union[List[DepsCall], DepsCall] = _DEFAULT_CONSTRAINT_VALUES["call_after"],
     ) -> None:
         self.language = language
         self.library_name = library_name
         self.function_name = function_name
         # Types must be stored as strings, since not all type objects can be pickled
         self.argtypes = [(arg[0].__name__, arg[1]) for arg in argtypes]
 
+        # Copied from electron.py
+        deps = {}
+
+        if isinstance(deps_bash, DepsBash):
+            deps["bash"] = deps_bash
+        if isinstance(deps_bash, list) or isinstance(deps_bash, str):
```

Comment:
will this not work for any list? should we check specifically for a list of strings?

Reply:
Carrying over discussion to https://github.com/AgnostiqHQ/covalent/pull/796